### PR TITLE
feat: tweak some CDK settings for rds and the ecs container

### PR DESCRIPTION
**Pull request by lart2150**

### CODE REVIEW

File: skeleton/templates/cdk/src/api-stack.ts.mustache [hunk 1]
```diff
@@ -67,6 +68,8 @@ export class ApiStack extends Stack {
                 enablePerformanceInsights: true,
             }),
             defaultDatabaseName: "api",
+            storageEncrypted: true,
+            readers: [],
```
Comment (DASPRiD) [inline skeleton/templates/cdk/src/api-stack.ts.mustache:72]: This shouldn't be required, as this should be the default, according to my tests.

File: skeleton/templates/cdk/src/api-stack.ts.mustache [hunk 2]
```diff
@@ -67,6 +68,8 @@ export class ApiStack extends Stack {
                 enablePerformanceInsights: true,
             }),
             defaultDatabaseName: "api",
+            storageEncrypted: true,
```
Comment (DASPRiD) [inline skeleton/templates/cdk/src/api-stack.ts.mustache:71]: Isn't Aurora encrypted in rest by default?

File: skeleton/templates/cdk/src/api-stack.ts.mustache [hunk 3]
```diff
@@ -162,8 +165,11 @@ export class ApiStack extends Stack {
             assignPublicIp: true,
             publicLoadBalancer: true,
             healthCheckGracePeriod: Duration.seconds(10),
+            sslPolicy: SslPolicy.RECOMMENDED_TLS,
```
Comment (DASPRiD) [inline skeleton/templates/cdk/src/api-stack.ts.mustache:168]: What exactly is this changing?

File: skeleton/templates/cdk/src/api-stack.ts.mustache [hunk 4]
```diff
@@ -162,8 +165,11 @@ export class ApiStack extends Stack {
...
         });
 
+        albService.loadBalancer.setAttribute('routing.http.drop_invalid_header_fields.enabled', 'true');
```
Comment (DASPRiD) [inline skeleton/templates/cdk/src/api-stack.ts.mustache:171]: Can you elaborate on these two values? What are they doing?

File: skeleton/templates/cdk/src/api-stack.ts.mustache (hunk 1 above)
Comment (lart2150) [reply skeleton/templates/cdk/src/api-stack.ts.mustache:72]: ~~I just spun up a stack and without readers it created a reader instance.~~

I take it back it didn't create a reader instance.  I'll remove the readers array.

File: skeleton/templates/cdk/src/api-stack.ts.mustache (hunk 2 above)
Comment (lart2150) [reply skeleton/templates/cdk/src/api-stack.ts.mustache:71]: it is not.

File: skeleton/templates/cdk/src/api-stack.ts.mustache (hunk 3 above)
Comment (lart2150) [reply skeleton/templates/cdk/src/api-stack.ts.mustache:168]: by default cdk sets this to the crummy setting of ELBSecurityPolicy-2016-08
https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/elb-security-policy-table.html

RECOMMENDED_TLS should use the recommended version for the currently used cdk version.

File: skeleton/templates/cdk/src/api-stack.ts.mustache (hunk 4 above)
Comment (lart2150) [reply skeleton/templates/cdk/src/api-stack.ts.mustache:171]: https://docs.aws.amazon.com/securityhub/latest/userguide/elb-controls.html#elb-4
* block invalid headers
https://docs.aws.amazon.com/securityhub/latest/userguide/elb-controls.html#elb-12
* block request smuggling

They are two security features built into the ALB that are free to enable and shouldn't impact legit traffic.

**Pull request merged**